var/spack/repos/builtin/packages/py-abipy/package.py

```diff
@@ -13,44 +13,65 @@ class PyAbipy(PythonPackage):
     homepage = "https://github.com/abinit/abipy"
     pypi = "abipy/abipy-0.2.0.tar.gz"
 
+    version("0.9.3", sha256="2f750c1ddcf78d0ec77c1bd44624ff96f80436307237589c02da127106514b75")
     version("0.2.0", sha256="c72b796ba0f9ea4299eac3085bede092d2652e9e5e8074d3badd19ef7b600792")
 
     variant("gui", default=False, description="Build the GUI")
-    variant("ipython", default=False, description="Build IPython support")
-
-    depends_on("python@2.7:")
+    variant("ipython", default=False, when="0.2.0", description="Build IPython support")
 
     depends_on("py-setuptools", type="build")
     depends_on("py-cython", type="build")
+    # in newer pip versions --install-option does not exist
+    depends_on("py-pip@:23.0", when="+ipython", type="build")
 
-    depends_on("py-six", type=("build", "run"))
-    depends_on("py-prettytable", type=("build", "run"))
+    depends_on("py-monty", when="@0.7:", type=("build", "run"))
     depends_on("py-tabulate", type=("build", "run"))
-    depends_on("py-apscheduler@2.1.0", type=("build", "run"))
+    depends_on("py-apscheduler", when="@0.9:", type=("build", "run"))
+    depends_on("py-apscheduler@2.1.0", when="@:0.8", type=("build", "run"))
     depends_on("py-pydispatcher@2.0.5:", type=("build", "run"))
     depends_on("py-tqdm", type=("build", "run"))
-    depends_on("py-html2text", type=("build", "run"))
     depends_on("py-pyyaml@3.11:", type=("build", "run"))
     depends_on("py-pandas", type=("build", "run"))
     depends_on("py-numpy@1.9:", type=("build", "run"))
     depends_on("py-scipy@0.14:", type=("build", "run"))
```

Comment:
Where do these lower bounds come from?

Reply:
The came from py-abipy@0.2.0 and I thought it does not hurt to leave them. But to avoid future confusion, I have removed them now.